File: go/scoreOfString.go
func scoreOfString(s string) int {
    result := 0;

    for i := 0; i < len(s); i++ {
        if (i < len(s) - 1) {
            currentValue := int(s[i]);
            previousValue := int(s[i + 1]);

            if currentValue > previousValue {
                result += currentValue - previousValue;
            } else {
                result += previousValue - currentValue;
            }
        }
    }

    return result;
}

File: typescript/scoreOfString.ts
function scoreOfString(s: string): number {
    let result = 0;

    for (let i = 0; i < s.length; i++) {
        if (i < s.length - 1) {
            const currentValue = s[i].charCodeAt(0);
            const previousValue = s[i + 1].charCodeAt(0);

            if (currentValue > previousValue) {
                result += currentValue - previousValue;
            } else {
                result += previousValue - currentValue;
            }
        }
    }

    return result;
};